include/nil/actor/zk/snark/arithmetization/plonk/assignment.hpp

```diff
@@ -129,12 +133,17 @@ namespace nil {
                         return _public_inputs[index].size();
                     }
 
-                    ColumnType public_input(std::uint32_t index) const {
+                    const ColumnType& public_input(std::uint32_t index) const {
                         assert(index < public_inputs_amount());
                         return _public_inputs[index];
                     }
 
-                    public_input_container_type public_inputs() const {
+                    // We need a way to access _public_inputs in a non-const form.
+                    public_input_container_type& public_inputs() {

```

Comment:
Here I added these functions, to make a few tests compile. Not sure if they are needed.